tutorial/python_agents/base_agent.py

```diff
@@ -0,0 +1,485 @@
+#!/usr/bin/env python3
+
+#  Copyright (c) 2015 - 2022, Intel Corporation
+#
+#  Redistribution and use in source and binary forms, with or without
+#  modification, are permitted provided that the following conditions
+#  are met:
+#
+#      * Redistributions of source code must retain the above copyright
+#        notice, this list of conditions and the following disclaimer.
+#
+#      * Redistributions in binary form must reproduce the above copyright
+#        notice, this list of conditions and the following disclaimer in
+#        the documentation and/or other materials provided with the
+#        distribution.
+#
+#      * Neither the name of Intel Corporation nor the names of its
+#        contributors may be used to endorse or promote products derived
+#        from this software without specific prior written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+#  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+#  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+#  A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+#  OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+#  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+#  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+#  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+#  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+#  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY LOG OF THE USE
+#  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+
+"""
+This package provides a BaseAgent class to monitor processes. Extend this class
+to implement additional agent functionality.
+
+The package provides parse_common_args and run_controller functions
+to create an argument parser for classes that extend BaseAgent.
+"""
+
+import sys
+import os
+import argparse
+from collections import OrderedDict
+import socket
+import yaml
+
+import geopmdpy
+import geopmdpy.runtime
+import geopmdpy.pio
+import geopmpy.reporter
+
+
+CSV_SEP_CHAR = "|"
+DOMAIN_SEP_CHAR = "-"
+TIME_SIGNAL = ("TIME", geopmdpy.topo.DOMAIN_BOARD, 0, "-")
+
+DEFAULT_SIGNAL_LIST = [
+    ("INSTRUCTIONS_RETIRED",         geopmdpy.topo.DOMAIN_BOARD,        0, "incr"),
+    ("POWER_PACKAGE",                geopmdpy.topo.DOMAIN_BOARD,        0, "avg"),
+    ("POWER_DRAM",                   geopmdpy.topo.DOMAIN_BOARD_MEMORY, 0, "avg"),
+    ("FREQUENCY",                    geopmdpy.topo.DOMAIN_BOARD,        0, "avg"),
+    ("MSR::UNCORE_PERF_STATUS:FREQ", geopmdpy.topo.DOMAIN_BOARD,        0, "avg"),
+    ("TEMPERATURE_CORE",             geopmdpy.topo.DOMAIN_BOARD,        0, "avg"),
+    ("ENERGY_DRAM",                  geopmdpy.topo.DOMAIN_BOARD_MEMORY, 0, "incr"),
+    ("ENERGY_PACKAGE",               geopmdpy.topo.DOMAIN_BOARD,        0, "incr"),
```

Comment:
```suggestion
    ("CPU_INSTRUCTIONS_RETIRED",           geopmdpy.topo.DOMAIN_BOARD, 0, "incr"),
    ("CPU_POWER",                                       geopmdpy.topo.DOMAIN_BOARD, 0, "avg"), 
    ("DRAM_POWER",                                   geopmdpy.topo.DOMAIN_BOARD, 0, "avg"),
    ("CPU_FREQUENCY_STATUS",                 geopmdpy.topo.DOMAIN_BOARD, 0, "avg"), 
    ("CPU_UNCORE_FREQUENCY_STATUS", geopmdpy.topo.DOMAIN_BOARD, 0, "avg"),  
    ("CPU_CORE_TEMPERATURE",                geopmdpy.topo.DOMAIN_BOARD, 0, "avg"),
    ("DRAM_ENERGY",                                 geopmdpy.topo.DOMAIN_BOARD, 0, "incr"),
    ("CPU_ENERGY",                                     geopmdpy.topo.DOMAIN_BOARD, 0, "incr"), 
```

WSP issues aside in the above suggestion these should be the new 2.0 names & aliases, tested on mcfly